### PR TITLE
fix: consider from date in SLE query

### PR DESCRIPTION
**Pull request by rtdany10**
In cases where from date isn't provided, the system throws error, but it doesn't apply the from date in the query if provided.

### CODE REVIEW

**Comment by rtdany10** [comment]
@s-aga-r please check

**Review by s-aga-r — APPROVED** [review]

**Comment by s-aga-r** [comment]
> @s-aga-r please check

@rtdany10 Add a test case for the same.

**Review by s-aga-r — CHANGES_REQUESTED** [review]
Add a test case for the same.

**Comment by rtdany10** [comment]
@s-aga-r I'm not sure why the existing test case is failing. Any pointers?

**Comment by s-aga-r** [comment]
> @s-aga-r I'm not sure why the existing test case is failing. Any pointers?

@rtdany10 not sure, but it can be because tests have from_filter and now it's applying in the query?

**Comment by rtdany10** [comment]
> > @s-aga-r I'm not sure why the existing test case is failing. Any pointers?
> 
> @rtdany10 not sure, but it can be because tests have from_filter and now it's applying in the query?

@s-aga-r  I'm not sure, cause there are no hard-coded values, comparison occurs between SLE values and report values. I'll try digging deeper.

**Comment by s-aga-r** [comment]
> > > @s-aga-r I'm not sure why the existing test case is failing. Any pointers?
> > 
> > 
> > @rtdany10 not sure, but it can be because tests have from_filter and now it's applying in the query?
> 
> @s-aga-r I'm not sure, cause there are no hard-coded values, comparison occurs between SLE values and report values. I'll try digging deeper.

Hi, @rtdany10 any update on this? Got a Support Ticket related to this.

File: erpnext/stock/report/stock_balance/test_stock_balance.py [hunk 1]
```diff
@@ -53,9 +52,11 @@ def assertInvariants(self, rows):
 						ORDER BY timestamp(posting_date, posting_time) desc, creation desc)
 						AS rn
 					FROM `tabStock Ledger Entry`
-					where is_cancelled=0
+					where posting_date >= %s and posting_date <= %s and is_cancelled=0
```
Comment (ankush) [inline erpnext/stock/report/stock_balance/test_stock_balance.py:55]: Never modify test code to them like to pass other tests! 

It is encoding a known supported business use case. 😅

File: erpnext/stock/report/stock_balance/stock_balance.py [hunk 2]
```diff
@@ -259,7 +259,9 @@ def apply_conditions(query, filters):
 	sle = frappe.qb.DocType("Stock Ledger Entry")
 	warehouse_table = frappe.qb.DocType("Warehouse")
 
-	if not filters.get("from_date"):
+	if from_date := filters.get("from_date"):
+		query = query.where(sle.posting_date >= from_date)
```
Comment (ankush) [inline erpnext/stock/report/stock_balance/stock_balance.py:263]: This is incorrect. Stock balance report has to fetch all SLEs, the from date is only used for computing opening qty. 

This is somewhat inefficient but that's how it's implemented. It's _correct_ at least.

**Review by ankush — CHANGES_REQUESTED** [review]

File: erpnext/stock/report/stock_balance/stock_balance.py (hunk 2 above)
Comment (ankush) [reply erpnext/stock/report/stock_balance/stock_balance.py:263]: @s-aga-r

Comment (ankush) [reply erpnext/stock/report/stock_balance/stock_balance.py:263]: also refer https://github.com/frappe/erpnext/issues/29183

**Pull request closed without merging**